api/all/src/test/java/io/opentelemetry/api/internal/ReadOnlyArrayMapTest.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.internal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class ReadOnlyArrayMapTest {
+
+  @Test
+  void equalsHashCode() {
```

Comment:
FYI: Guava has a nice testing util for equal and hash.

Reply:
I didn't see anything for hashcode, but only the EqualsTester. Did I miss something for hashcode?

Reply:
EqualTester tests for both and all possible combinations that you give see https://www.javadoc.io/doc/com.google.guava/guava-testlib/21.0/com/google/common/testing/EqualsTester.html